# Bound state core ionization of atoms using CS-RHF

In [ ]:
import numpy as np
from py_mods.src.SCF.CSUHF import CS_UHF
from py_mods.src.SCF.external import UHF_context_from_pyscf

## $He$ $1s2s$ and $He^{-}$ $1s2s^{2}$

In [2]:
He_1s2s_context = UHF_context_from_pyscf('He 0 0 0', 0, 0, basis='aug-cc-pvqz')
He_1s2s_context.mult = 2
He_1s2s_context.occupation = (np.array([1,1]), np.array([0,0]))
# He_1s2s_context.verbose = True

In [3]:
He_1s2s_res = CS_UHF(He_1s2s_context)
E_ref_He_min_1s2s = -2.1743
print(f'${He_1s2s_res.E_UHF.real:.4f}$ | ${He_1s2s_res.E_UHF.real - E_ref_He_min_1s2s :.4f}$ |')


$-2.1712$ | $0.0031$ |


In [4]:
He_min_1s2s_context = UHF_context_from_pyscf('He 0 0 0', 0, 0, basis='aug-cc-pvqz')
He_min_1s2s_context.mult = 1
He_min_1s2s_context.n_electrons = 3
He_min_1s2s_context.conv_type = None
He_min_1s2s_context.occupation = (np.array([0,1]), np.array([1,1]))
# He_min_1s2s_context.verbose = True

In [5]:
He_min_1s2s_res = CS_UHF(He_min_1s2s_context) 
E_ref_He_min_1s2s2 = -2.1633
print(f'${He_min_1s2s_res.E_UHF.real:.4f}$ | ${He_min_1s2s_res.E_UHF.real - E_ref_He_min_1s2s2 :.4f}$ |')

$-2.1027$ | $0.0606$ |


## $Ne$ $2p^{5}3s$ and $Ne^-$ $2p^{5}3s^{2}$

In [6]:
Ne_2p5_3s_context = UHF_context_from_pyscf('Ne 0 0 0', 0, 0, basis='aug-cc-pvqz')
Ne_2p5_3s_context.mult = 2
Ne_2p5_3s_context.occupation = (np.array([1,1,1,1,1]), np.array([1,1,1,1,0,1]))
# Ne_2p5_3s_context.verbose = True

In [7]:
Ne_2p5_3s_res = CS_UHF(Ne_2p5_3s_context)
E_ref_Ne_2p5_3s = -127.9923
print(f'${Ne_2p5_3s_res.E_UHF.real:.4f}$ | ${Ne_2p5_3s_res.E_UHF.real - E_ref_Ne_2p5_3s :.4f}$ |')

$-127.8676$ | $0.1247$ |


In [8]:
Ne_min_2p5_3s2_context = UHF_context_from_pyscf('Ne 0 0 0', 0, 0, basis='aug-cc-pvqz')
Ne_min_2p5_3s2_context.mult = 1
Ne_min_2p5_3s2_context.occupation = (np.array([1,1,1,1,1,1]), np.array([1,1,1,1,0,1]))
# Ne_min_2p5_3s2_context.verbose = True
Ne_min_2p5_3s2_context.n_electrons = 11

In [9]:
Ne_min_2p5_3s2_res = CS_UHF(Ne_min_2p5_3s2_context)
E_ref_Ne_min_2p5_3s2 = -127.9867
print(f'${Ne_min_2p5_3s2_res.E_UHF.real:.4f}$ | ${Ne_min_2p5_3s2_res.E_UHF.real - E_ref_Ne_min_2p5_3s2 :.4f}$ |')

$-127.7466$ | $0.2401$ |


## $Ar$ $3p^{5}4s$ and $Ar^{-}$ $3p^{5}4s^{2}$

In [10]:
# Ar_3p5_4s_context = UHF_context_from_pyscf('Ar 0 0 0', 0, 0, basis='aug-cc-pvqz')
# Ar_3p5_4s_context.verbose = True
# Ar_3p5_4s_res = CS_UHF(Ar_3p5_4s_context)

In [11]:
Ar_3p5_4s_context = UHF_context_from_pyscf('Ar 0 0 0', 0, 0, basis='aug-cc-pvqz')
Ar_3p5_4s_context.mult = 2
Ar_3p5_4s_context.occupation = (np.array([1,1,1,1,1,1,1,1,1]), np.array([1,1,1,1,1,1,1,1,0,1]))
# Ar_3p5_4s_context.verbose = True
Ar_3p5_4s_context.break_symm = True
Ar_3p5_4s_context.conv_ITER_START = 30

In [12]:
Ar_3p5_4s_res = CS_UHF(Ar_3p5_4s_context)
E_ref_Ar_3p5_4s = -526.4167
print(f'${Ar_3p5_4s_res.E_UHF.real:.4f}$ | ${Ar_3p5_4s_res.E_UHF.real - E_ref_Ar_3p5_4s :.4f}$ |')

$-526.3568$ | $0.0599$ |


In [13]:
Ar_min_3p5_4s2_context = UHF_context_from_pyscf('Ar 0 0 0', 0, 0, basis='aug-cc-pvqz')
Ar_min_3p5_4s2_context.mult = 1
Ar_min_3p5_4s2_context.n_electrons = 19
Ar_min_3p5_4s2_context.occupation = (np.array([1,1,1,1,1,1,1,1,1,1]), np.array([1,1,1,1,1,1,1,1,0,1]))
# Ar_min_3p5_4s2_context.verbose = True
Ar_min_3p5_4s2_context.break_symm = True
Ar_min_3p5_4s2_context.conv_ITER_START = 30

In [14]:
Ar_min_3p5_4s2_res = CS_UHF(Ar_min_3p5_4s2_context)
E_ref_Ar_min_3p5_4s2 = -526.4116
print(f'${Ar_min_3p5_4s2_res.E_UHF.real:.4f}$ | ${Ar_min_3p5_4s2_res.E_UHF.real - E_ref_Ar_min_3p5_4s2 :.4f}$ |')

$-526.2870$ | $0.1246$ |


# Results 
These values are compared with [J. Chem. Phys. 52, 4363–4368 (1970)](https://doi.org/10.1063/1.1673658):

|State                  | Reference value | Calculated | Difference|
|-----------------------|-----------------|------------|-----------|
|$He$ $1s2s$            |   $-2.1743$ |   $-2.1712$ | $0.0031$ |
|$He^{-}$ $1s2s^{2}$    |   $-2.1633$ |   $-2.1027$ | $0.0606$ |
|$Ne$ $2p^{5}3s$        | $-127.9923$ | $-127.8676$ | $0.1247$ |
|$Ne^-$ $2p^{5}3s^{2}$  | $-127.9867$ | $-127.7466$ | $0.2401$ |
|$Ar$ $3p^{5}4s$        | $-526.4167$ | $-526.3568$ | $0.0599$ |
|$Ar^{-}$ $3p^{5}4s^{2}$| $-526.4116$ | $-526.2870$ | $0.1246$ |


